cylc/flow/async_util.py

```diff
@@ -478,3 +479,48 @@ async def _fcn(*args, executor=None, **kwargs):
 
 
 async_listdir = make_async(os.listdir)
+
+
+@asynccontextmanager
+async def async_block():
+    """Ensure all tasks started within the context are awaited when it closes.
+
+    Normally, you would await a task e.g:
+
+    await three()
+
+    If it's possible to await the task, do that, however, this isn't always an
+    option. This interface exists is to help patch over issues where async code
+    (one) calls sync code (two) which calls async code (three) e.g:
+
+    async def one():
+        two()
+
+    def two():
+        # this breaks - event loop is already running
+        asyncio.get_event_loop().run_until_complete(three())
+
+    async def three():
+        await asyncio.sleep(1)
+
+    This code will error because you can't nest asyncio (without nest-asyncio)
+    which means you can schedule tasks the tasks in "two", but you can't await
+    them.
+
+    def two():
+        # this works, but it doesn't wait for three() to complete
+        asyncio.create_task(three())
```

Comment:
According to the [docs ](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task)this usage may be risky.

Reply:
What is the safety issue? I can't see anything.

Also, if this is unsafe, then the entire approach is unsafe, not just the docs!

Reply:
https://github.com/python/cpython/issues/88831

Reply:
The issue is metomi-rose side, this commit attempts to solve it: https://github.com/metomi/rose/pull/2744/files#r1473137433